### PR TITLE
Initial implementation for metrics using PMD

### PR DESCRIPTION
**Pull request by bmbferreira**

### CODE REVIEW

File: src/main/scala/codacy/metrics/PMD.scala [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    }
+  }
+
+  private def calculateComplexity(directory: String, files: Option[Set[Source.File]]): Try[List[FileMetrics]] = {
+
+    val pmdConfig = buildConfig(directory, files)
+
+    val ruleSetFactory = RulesetsFactoryUtils.getRulesetFactory(pmdConfig, new ResourceLoader())
+
+    val languages = new java.util.HashSet[net.sourceforge.pmd.lang.Language]
```
Comment (xplosunn) [inline src/main/scala/codacy/metrics/PMD.scala]: You're already importing the "pmd" package, so you can just have "pmd.lang.Language".

File: src/main/scala/codacy/metrics/PMD.scala [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    languages.add(new JavaLanguageModule())
+
+    val applicableFiles = pmd.PMD.getApplicableFiles(pmdConfig, languages)
+
+    Try {
+      val codacyRenderer = new CodacyInMemoryRenderer()
+      val renderer: Renderer = codacyRenderer
```
Comment (xplosunn) [inline src/main/scala/codacy/metrics/PMD.scala]: This line seems useless. You can just add the type annotation to the line above if you need it.

File: src/main/scala/codacy/metrics/PMD.scala [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+          </properties>
+        </rule>
+      </ruleset>
+    val config = fileForConfig(xmlConfiguration)
+
+    config.map(p => pmdConfig.setRuleSets(p.toString))
+    pmdConfig
+  }
+
+  private def fileForConfig(config: Elem) = tmpfile(config)
```
Comment (rtfpessoa) [inline src/main/scala/codacy/metrics/PMD.scala]: Why do you need this method?

File: src/main/scala/codacy/metrics/PMD.scala [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        <description>Codacy UI Ruleset</description>
+        <rule ref="category/java/design.xml/CyclomaticComplexity">
+          <properties>
+            <property name="methodReportLevel" value="1"/>
+          </properties>
+        </rule>
+      </ruleset>
+    val config = fileForConfig(xmlConfiguration)
+
+    config.map(p => pmdConfig.setRuleSets(p.toString))
```
Comment (rtfpessoa) [inline src/main/scala/codacy/metrics/PMD.scala]: Shouldn't we throw a pretty error if we cannot get a config?

**Review by rtfpessoa — CHANGES_REQUESTED** [review]

**Review by rtfpessoa — APPROVED** [review]

**Pull request merged**